### PR TITLE
Fixes for flake8 linting

### PR DESCRIPTION
**Pull request by rajithkrishnegowda**
This PR addresses and fixes the linting issues identified by flake8 to ensure that the Ubuntu (latest) pipeline runs successfully.

**Fixes: Ubuntu (latest) pipeline**
https://github.com/securefederatedai/openfl/actions/runs/11656210544/job/32452012070

Changes:

- Updated various files to comply with flake8 linting rules.
- Ensured consistent code formatting and improved readability.

Tested locally

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+    dist,
+    .venv
 
-max-line-length = 100
+max-line-length = 120
 
 copyright-check = True
+
+# Enable specific checks or plugins
+select = B,C,E,F,W,T4,B9
```
Comment (MasterSkepticista) [inline setup.cfg:39]: Didn't understand, what does this line do?

File: setup.cfg [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
 exclude = 
     *_pb2*,
+    .git,
+    __pycache__,
+    build,
+    dist,
+    .venv
 
-max-line-length = 100
+max-line-length = 120
```
Comment (MasterSkepticista) [inline setup.cfg]: Any specific reason to increase?

**Review by MasterSkepticista — COMMENTED** [review]
@rajithkrishnegowda Also consider merging #1113 first. One of our former engineers opened it addressing Flake8. This PR can be incremental on changes from #1113.

File: setup.cfg (hunk 2 above)
Comment (rajithkrishnegowda) [reply setup.cfg]: standard max-line-length used from 79 or 120. so increased to max i.e. 120. if not needed we can revert it

File: setup.cfg (hunk 1 above)
Comment (rajithkrishnegowda) [reply setup.cfg:39]: select specifies specific error and warning codes or plugin checks to enable:

B: Bandit security checks (e.g., detecting insecure function use).
C: Cyclomatic complexity, used to flag overly complex functions.
E: PEP8 errors (e.g., style issues).
F: Pyflakes errors, like unused imports or undefined names.
W: PEP8 warnings (e.g., stylistic issues).
T4: Type checking from third-party tools (like mypy).
B9: Bugbear, for additional warnings about potentially error-prone code.

Comment (MasterSkepticista) [reply setup.cfg:39]: It would be useful to have this comment block in `setup.cfg`.

**Review by tanwarsh — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (MasterSkepticista) [reply setup.cfg]: Let us not go beyond 100.
Initial one-shot formatting of this codebase took a lot of effort to bring down the avg from 120 to 100. Most standards say 80 (for historical reasons), while 100 is a reasonable default in major repositories.

Comment (MasterSkepticista) [reply setup.cfg]: If 100 returns lint errors, it is better to trim those lines to be under 100 chars instead.

Comment (rajithkrishnegowda) [reply setup.cfg]: reverted to 100

File: setup.cfg (hunk 1 above)
Comment (rajithkrishnegowda) [reply setup.cfg:39]: added comments in setup.cfg

**Review by teoparvanov — APPROVED** [review]
I've merged https://github.com/securefederatedai/openfl/pull/1113, as suggested by @MasterSkepticista. Please consider rebasing this PR's branch and merge the remaining deltas (if any).

**Pull request merged**